src/compile/scale/parse.ts

```diff
@@ -65,3 +66,27 @@ export function parseScale(model: UnitModel, channel: Channel) {
   }
   return scaleComponent;
 }
+
+export function moveSharedScaleUp(model: Model, scaleComponent: Dict<ScaleComponent>, child: Model, channel: Channel) {
+  // Check whether the scales are actually compatible, e.g. use the same sort or throw error
+
+  const childScale = child.component.scales[channel];
+  let modelScale = scaleComponent[channel];
+
+  if (modelScale) {
+    modelScale.domain = unionDomains(modelScale.domain, childScale.domain);
```

Comment:
Add `// TODO: merge properties if they are not conflicting`

(Beyond the scope of this PR.)